src/main/java/hudson/plugins/promoted_builds/PromotionPermissionHelper.java

```diff
@@ -40,19 +41,19 @@
 @Restricted(NoExternalUse.class)
 public class PromotionPermissionHelper {
 
-    public static void checkPermission(@Nonnull AbstractProject<?,?> target, @CheckForNull ManualCondition associatedCondition) {
-        if (!hasPermission(target, associatedCondition)) {
+    public static void checkPermission(@Nonnull AbstractProject<?,?> target, AbstractBuild<?,?> build, @CheckForNull ManualCondition associatedCondition) {
+        if (!hasPermission(target, build, associatedCondition)) {
             // TODO: Give a more accurate error message if the user has Promotion.PROMOTE but is not in the list of approvers.
             throw new AccessDeniedException2(Jenkins.getAuthentication(), Promotion.PROMOTE);
         }
     }
 
-    public static boolean hasPermission(@Nonnull AbstractProject<?,?> target, @CheckForNull ManualCondition associatedCondition) {
+    public static boolean hasPermission(@Nonnull AbstractProject<?,?> target, AbstractBuild<?,?> build, @CheckForNull ManualCondition associatedCondition) {
```

Comment:
It would be good to add either `@CheckForNull` or `@Nonnull` to the parameter for clarity (same for the other places where a parameter is added).